Release v49.0.0

hamudi-ahmed opened the pull request:
This release adds support for expo 49

hamudi-ahmed committed:
bump dependencies

hamudi-ahmed committed:
bump latest supported version in CLI

hamudi-ahmed committed:
upgrade test fixture to expo 49

hamudi-ahmed committed:
revert netinfo dependency to v9.3.10

hamudi-ahmed committed:
remove expo-modules-core resolution override from test fixture

This is no longer needed

hamudi-ahmed committed:
revert jest version bump and remove jest-expo dependency

hamudi-ahmed committed:
reinstate jest-expo (still currently disabled)

hamudi-ahmed committed:
bump bugsnag deps

hamudi-ahmed committed:
add config plugin to test fixture to allow http traffic on android

hamudi-ahmed committed:
Merge pull request #139 from bugsnag/PLAT-10553-support-v49

hamudi-ahmed committed:
update changelog

kennethlynne commented on package.json:
```diff
@@ -1,26 +1,26 @@
+    "jest-expo": "^48.0.1",
+    "lerna": "^6.0.1",
+    "react": "18.2.0",
+    "react-native": "0.72.1",
```
Can be bumped to 0.72.3

kennethlynne commented on features/fixtures/test-app/package.json:
```diff
@@ -9,20 +9,17 @@
-  },
-  "resolutions": {
-    "expo-modules-core": "1.2.3"
+    "react-native": "0.72.1"
```
Can be bumped to 0.72.3

hamudi-ahmed committed:
replace deprecated `Constants.manifest`

Constants.manifest is deprecated in SDK 49 and now emits a warning when accessed. Switch to using Constants.expoConfig which provides a consistent way to access the app config whether its an embedded build, Classic update or EAS update

hamudi-ahmed committed:
add unit tests for default releaseStage config

hamudi-ahmed committed:
bump react-native to 0.72.3

hamudi-ahmed committed:
Merge pull request #141 from bugsnag/replace-constants-manifest
